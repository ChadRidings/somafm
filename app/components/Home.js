import React, { Component, PropTypes } from 'react';
import Nav from './common/Nav';
import SideNav from './common/SideNav';
import styles from './Home.module.css';
import ChannelCard from './ChannelCard';
import SomaFMService from '../services/SomaFMService';

export default class Home extends Component {

  static propTypes = {
    setHomeChannels: PropTypes.func,
    channels: PropTypes.shape({
      home: PropTypes.arrayOf(PropTypes.shape({})),
      favorites: PropTypes.arrayOf(PropTypes.shape({}))
    })
  };

  componentDidMount() {
    this.getAllChannels();
    this.loadFavorites();
  }

  getAllChannels() {
    const soma = new SomaFMService();
    soma.getAllChannels((err, data) => {
      this.props.setHomeChannels(data);
    });
  }

  loadFavorites() {
    const soma = new SomaFMService();
    soma.loadSavedChannels((data) => {
      this.props.setFavorites(data);
    });
  }

  render() {
    const channels = this.props.channels.home || [];
    const channelNodes = channels.map((v, i) =>
      <ChannelCard
        key={i}
        img={v.largeimage || "http://placehold.it/400x300"}
        title={v.title}
        subtitle={v.description}
        url={v.id}
      />
    );

    return (
      <div className={styles.home}>
        <SideNav favorites={this.props.channels.favorites} />
        <div className={styles.container}>
          <Nav />
          <div className="channels">
            <h2>Channels</h2>
            {channels.length !== 0 ? channelNodes : 'No channels available.' }
          </div>
        </div>
      </div>
    );
  }
}
