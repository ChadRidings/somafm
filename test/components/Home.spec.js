import { spy } from 'sinon';
import React from 'react';
import { shallow } from 'enzyme';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Home from '../../app/components/Home';

const channels = {
  favorites: [],
  home: [
    {
      id: '7soul',
      title: 'Seven Inch Soul',
      description: 'Vintage soul tracks from the original 45 RPM vinyl.',
      dj: 'Dion Watts Garcia',
      djmail: 'dion@somafm.com',
      genre: 'oldies',
      image: 'https://api.somafm.com/img/7soul120.png',
      largeimage: 'https://api.somafm.com/logos/256/7soul256.png',
      xlimage: 'https://api.somafm.com/logos/512/7soul512.png',
      twitter: 'SevenInchSoul',
      updated: '1396144686',
      playlists: [
        {
          url: 'https://api.somafm.com/7soul130.pls',
          format: 'aac',
          quality: 'highest'
        },
        {
          url: 'https://api.somafm.com/7soul.pls',
          format: 'mp3',
          quality: 'high'
        },
        {
          url: 'https://api.somafm.com/7soul64.pls',
          format: 'aacp',
          quality: 'high'
        },
        {
          url: 'https://api.somafm.com/7soul32.pls',
          format: 'aacp',
          quality: 'low'
        }
      ],
      listeners: '53',
      lastPlaying: "Bobby Freeman - C'mon And Swim"
    }
  ]
};

function setup() {
  const actions = {
    // increment: spy(),
    // incrementIfOdd: spy(),
    // incrementAsync: spy(),
    // decrement: spy()
    getHomeChannels: spy()
  };
  const component = shallow(<Home channels={channels} {...actions} />);
  return {
    component,
    actions,
    channelCards: component.find('.channelCard'),
    p: component.find('.channels')
  };
}

describe('Home component', () => {
  it('should should display channel header', () => {
    const { p } = setup();
    expect(p.find('h2').text()).toMatch('Channels');
  });

  // it('should first button should call increment', () => {
  //   const { buttons, actions } = setup();
  //   buttons.at(0).simulate('click');
  //   expect(actions.increment.called).toBe(true);
  // });

  it('should match exact snapshot', () => {
    const { actions } = setup();
    const tree = renderer
      .create(
        <div>
          <Router>
            <Home channels={channels} {...actions} />
          </Router>
        </div>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  // it('should second button should call decrement', () => {
  //   const { buttons, actions } = setup();
  //   buttons.at(1).simulate('click');
  //   expect(actions.decrement.called).toBe(true);
  // });

  // it('should third button should call incrementIfOdd', () => {
  //   const { buttons, actions } = setup();
  //   buttons.at(2).simulate('click');
  //   expect(actions.incrementIfOdd.called).toBe(true);
  // });

  // it('should fourth button should call incrementAsync', () => {
  //   const { buttons, actions } = setup();
  //   buttons.at(3).simulate('click');
  //   expect(actions.incrementAsync.called).toBe(true);
  // });
});
