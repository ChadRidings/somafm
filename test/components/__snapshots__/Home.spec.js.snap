// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component should match exact snapshot 1`] = `
<div>
  <div
    className="home"
  >
    <div
      className="sideNav"
    >
      <a
        className="logo"
        href="/"
        onClick={[Function]}
      />
      <ul
        className="navLinks"
      >
        <li
          className="title"
        >
          Main
        </li>
        <a
          className="active"
          href="/channels"
          onClick={[Function]}
        >
          <li>
            <i
              className="fa fa-hashtag"
            />
             Channels
          </li>
        </a>
      </ul>
      <ul
        className="navLinks"
      >
        <li
          className="title"
        >
          Favorites
        </li>
      </ul>
      <ul
        className="status"
      >
        <li>
          <i
            className="fa fa-signal"
          />
           
          <span
            className="offline"
          >
            offline
          </span>
        </li>
      </ul>
    </div>
    <div
      className="container"
    >
      <div
        className="nav"
        onScroll={[Function]}
        style={undefined}
      >
        <ul
          className="stats"
        />
      </div>
      <div
        className="channels"
      >
        <h2>
          Channels
        </h2>
        <div
          className="channelCard channelCard"
        >
          <a
            href="/channel/7soul"
            onClick={[Function]}
          >
            <img
              alt="Seven Inch Soul"
              className="channelThumbnail__img"
              src="https://api.somafm.com/logos/256/7soul256.png"
            />
          </a>
          <div
            className="channelCardContent"
          >
            <div
              className="channelCardContent__title"
            >
              Seven Inch Soul
            </div>
            <div
              className="channelCardContent__subtitle"
            >
              Vintage soul tracks from the original 45 RPM vinyl.
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
